va/http.go

```diff
@@ -539,19 +547,11 @@ func (va *ValidationAuthorityImpl) processHTTPValidation(
 		// assign to the client transport in order to connect to the redirect target using
 		// the IP address we selected.
 		redirDialer, redirRecord, err := va.setupHTTPValidation(ctx, req.URL.String(), redirTarget)
+		records = append(records, redirRecord)
 		if err != nil {
 			return err
 		}
```

Comment:
```suggestion
		if err != nil {
			return err
		}
		records = append(records, redirRecord)
```

Even though two lines might seem like an atomic operation, always best to check the error code before doing anything else.

Reply:
Can do. I was putting that bit of code back the way I found it, but can add that to the changeset.

Reply:
Thanks! Fixed. I had put that bit of code back the way I found it, but this is better.

Reply:
I believe this is actually intentional, because `setupHTTPValidation` can error with, e.g. a DNS lookup failure, and we want that to go on the list of validation records. We should have a unittest to cover that case, but for now I'd like to leave the order of statements as it is.

Reply:
Thanks @jsha. Reverted for now and will open an issue to make sure we are testing for that condition.

Reply:
Ah, I see. Yeah, I think that this is indicative of a code smell: the fact that that inner function can error out but the contents of `records` are still relevant (because the function is a closure over `records`) is spooky-action-at-a-distance. Especially when the closure is this long (60 lines!) it's easy to lose track of things like this.

Leaving it as we found it is definitely fine for this change, but it would be good to take a look at potential refactorings here in the future.